yarn/src/main/scala/org/apache/spark/deploy/yarn/security/HDFSCredentialProvider.scala

```diff
@@ -75,8 +84,15 @@ private[security] class HDFSCredentialProvider extends ServiceCredentialProvider
     sparkConf.get(PRINCIPAL).flatMap { renewer =>
       val creds = new Credentials()
       nnsToAccess(hadoopConf, sparkConf).foreach { dst =>
-        val dstFs = dst.getFileSystem(hadoopConf)
-        dstFs.addDelegationTokens(renewer, creds)
+          try {
+            val dstFs = dst.getFileSystem(hadoopConf)
+            dstFs.addDelegationTokens(renewer, creds)
+          } catch {
+            case e: StandbyException =>
+              logWarning(s"Namenode ${dst} is in state standby", e)
+            case e: RemoteException =>
+              logWarning(s"Namenode ${dst} is in state standby", e)
+          }
```

Comment:
HADOOP-13372 implies that swift:// throws an UnknownHostException here; best to catch & log too, in case someone adds swift:// to the list of filesystems.